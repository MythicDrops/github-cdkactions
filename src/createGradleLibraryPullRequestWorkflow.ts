import { Job, Stack, StepsProps, Workflow } from "cdkactions";
import dedent from "ts-dedent";

export const createGradleLibraryPullRequestWorkflow = (
  stack: Stack,
  preTestSteps: StepsProps[] = []
): Workflow => {
  const workflow = new Workflow(stack, "pull-request", {
    name: "Pull Request",
    on: {
      pullRequest: { branches: ["main"] },
      push: { branchesIgnore: ["main"] },
    },
  });
  new Job(workflow, "pull-request", {
    name: "CI",
    runsOn: "ubuntu-latest",
    steps: [
      {
        name: "Check-out Code",
        uses: "actions/checkout@v2",
        with: { "fetch-depth": 0 },
      },
      {
        name: "Set up JDK",
        uses: "actions/setup-java@v1",
        with: { "java-version": "1.8" },
      },
      {
        name: "Make Gradle Wrapper Executable",
        run: dedent`chmod +x ./gradlew`,
      },
      {
        name: "Set Gradle Version",
        run: dedent`./gradlew wrapper --gradle-version 6.8.2`,
      },
      ...preTestSteps, // expand preTestSteps
      {
        name: "Test Library with Gradle",
        run: dedent`./gradlew check`,
      },
    ],
  });
  return workflow;
};
