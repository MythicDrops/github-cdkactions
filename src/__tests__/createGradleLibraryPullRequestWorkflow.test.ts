import { App, Stack } from "cdkactions";
import { createGradleLibraryPullRequestWorkflow } from "../createGradleLibraryPullRequestWorkflow";

describe("createGradleLibraryPullRequestWorkflow", () => {
  let app: App;
  let stack: Stack;

  beforeEach(() => {
    app = new App();
    stack = new Stack(app, "cdk");
  });

  it("generates a GitHub Actions workflow", () => {
    expect(
      createGradleLibraryPullRequestWorkflow(stack).toGHAction()
    ).toMatchSnapshot();
  });
});
