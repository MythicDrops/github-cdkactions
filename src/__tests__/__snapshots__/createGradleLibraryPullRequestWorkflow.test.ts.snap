// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGradleLibraryPullRequestWorkflow generates a GitHub Actions workflow 1`] = `
Object {
  "jobs": Object {
    "pull-request": Object {
      "name": "CI",
      "runs-on": "ubuntu-latest",
      "steps": Array [
        Object {
          "name": "Check-out Code",
          "uses": "actions/checkout@v2",
          "with": Object {
            "fetch-depth": 0,
          },
        },
        Object {
          "name": "Set up JDK",
          "uses": "actions/setup-java@v2",
          "with": Object {
            "distribution": "adopt",
            "java-version": "8",
          },
        },
        Object {
          "name": "Make Gradle Wrapper Executable",
          "run": "chmod +x ./gradlew",
        },
        Object {
          "name": "Set Gradle Version",
          "run": "./gradlew wrapper --gradle-version 7.0.2",
        },
        Object {
          "name": "Test Library with Gradle",
          "run": "./gradlew check",
        },
        Object {
          "env": Object {
            "CODECOV_TOKEN": "\${{ secrets.CODECOV_TOKEN }}",
          },
          "name": "Upload Code Coverage to codecov",
          "run": "bash <(curl -s https://codecov.io/bash)",
        },
      ],
    },
  },
  "name": "Pull Request",
  "on": Object {
    "pull_request": Object {
      "branches": Array [
        "main",
      ],
    },
    "push": Object {
      "branches-ignore": Array [
        "main",
      ],
    },
  },
}
`;
